docs/structured-streaming-programming-guide.md

```diff
@@ -1647,6 +1648,26 @@ For example, sorting on the input stream is not supported, as it requires keepin
 track of all the data received in the stream. This is therefore fundamentally hard to execute 
 efficiently.
 
+### Limitation of global watermark
+
+In Append mode, some stateful operations could emit rows older than current watermark plus allowed late record delay,
+which are "late rows" in downstream stateful operations (as Spark uses global watermark) and these rows can be discarded.
```

Comment:
can be discarded implies that it's OK to discard them. Are you saying "may be discarded"?
Then I'd say "if a stateful operation emits rows older ... note that these rows may be discarded"

Reply:
> can be discarded implies that it's OK to discard them. Are you saying "may be discarded"?

Ah thanks for correcting the details. If former interprets as that nuance, latter is correct.